app/models/wpcc/your_contribution.rb

```diff
@@ -4,5 +4,20 @@ class YourContribution
     attr_accessor :eligible_salary,
                   :employee_percent,
                   :employer_percent
+
+    CONTRIBUTION_PREFERENCES = {
+      'minimum' => MinimumContributionCalculator,
+      'full' => FullContributionCalculator
+    }.freeze
+
+    def self.generate(salary, contribution_preference)
+      contribution_klass = CONTRIBUTION_PREFERENCES[contribution_preference]
+      calculator = contribution_klass.new(salary, contribution_preference)
+      new(
+        eligible_salary:  calculator.eligible_salary,
+        employee_percent: calculator.employee_percent,
+        employer_percent: calculator.employer_percent
+      )
+    end
```

Comment:
I like this.. but I also agree with @margOnline and think it could be made a bit clearer..
I'm still thinking why exactly it's niggling with me though.. and what a valid alternative suggestion would be.